ts/src/header-validator/validate-json.ts

```diff
@@ -1542,3 +1565,112 @@ export function validateTrigger(
     trigger
   )
 }
+
+export type EventLevelReport = {
+  attributionDestination: string | string[]
+  randomizedTriggerRate: number
+  reportId: string
+  scheduledReportTime: bigint
+  sourceDebugKey: bigint | null
+  sourceType: SourceType
+  sourceEventId: bigint
+  triggerData: bigint
+  triggerDebugKey: bigint | null
+  triggerSummaryBucket: [number, number] | null
+}
+
+function reportDestination(ctx: Context, j: Json): Maybe<string | string[]> {
+  const suitableSiteNoExtraneous = (ctx: Context, j: Json) =>
+    suitableSite(ctx, j, /*rejectExtraComponents=*/ true)
+
+  return typeSwitch<string | string[]>(ctx, j, {
+    string: (ctx, j) => suitableSiteNoExtraneous(ctx, j),
+    list: (ctx, j) =>
+      array(ctx, j, suitableSiteNoExtraneous, { minLength: 1, maxLength: 3 }),
+  })
+}
+
+function randomizedTriggerRate(ctx: Context, j: Json): Maybe<number> {
+  return number(ctx, j).filter((n) => isInRange(ctx, n, 0, 1))
+}
+
+function randomUuid(ctx: Context, j: Json): Maybe<string> {
+  return string(ctx, j).filter((s) => {
+    if (!uuid.validate(s)) {
+      ctx.error('must be a valid UUID')
+      return false
+    }
+    if (uuid.version(s) !== 4) {
+      ctx.error('must be a version 4 (random) UUID')
+      return false
+    }
+    return true
+  })
+}
+
+function triggerSummaryBucket(ctx: Context, j: Json): Maybe<[number, number]> {
+  let prev = 1
+  let prevDesc = 'minimum bucket start'
+
+  const endpoint = (ctx: Context, j: Json): Maybe<number> =>
+    number(ctx, j)
+      .filter((n) => isInteger(ctx, n))
+      .filter((n) =>
+        isInRange(
+          ctx,
+          n,
+          prev,
+          UINT32_MAX,
+          `must be >= ${prevDesc} (${prev}) and <= uint32 max (${UINT32_MAX})`
+        )
+      )
+      .peek((n) => {
+        prev = n
+        prevDesc = 'bucket start'
+      })
+
+  return array(ctx, j, endpoint, {
+    minLength: 2,
+    maxLength: 2,
+    keepGoing: false,
+  }) as Maybe<[number, number]>
+}
+
+function eventLevelReport(
+  ctx: GenericContext,
+  j: Json
+): Maybe<EventLevelReport> {
+  return object(ctx, j).map((j) => {
+    return struct(ctx, j, {
+      attributionDestination: field(
+        'attribution_destination',
+        reportDestination
+      ),
+      randomizedTriggerRate: field(
+        'randomized_trigger_rate',
+        randomizedTriggerRate
+      ),
+      reportId: field('report_id', randomUuid),
+      scheduledReportTime: field('scheduled_report_time', int64),
```

Comment:
Do we need test overage for types of these fields?

Reply:
Done.